### PR TITLE
feat: additional APIs to support "actual select all" functions [ET-238]

### PR DESCRIPTION
**Pull request by jesse-amano-hpe**
## Ticket
Supports [ET-238] with resources we missed in planning

## Description
Fixes a couple bugs in bulk `KillSearches`, adds `CancelSearches` and `LaunchTensorboardSearches`

## Test Plan
With the rest of [ET-238], user should be able to select all searches matching a filter in the Web UI with a "true select all button", then perform bulk actions including "cancel" and "launch Tensorboard" and have them apply to all searches that match the filter even if they are not displayed on the current page view.

## Checklist

- [x] Changes have been manually QA'd
- [x] New features have been approved by the corresponding PM
- [x] User-facing API changes have the "User-facing API Change" label
- [x] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses have been included for new code which was copied and/or modified from any external code

[ET-238]: https://hpe-aiatscale.atlassian.net/browse/ET-238?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ET-238]: https://hpe-aiatscale.atlassian.net/browse/ET-238?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by EmilyBonar — APPROVED** [review]
Thanks for the assist!

File: master/internal/api_searches.go [hunk 1]
```diff
@@ -210,80 +210,40 @@ func (a *apiServer) MoveSearches(
...
-	visibleIDs := set.New[int32]()
-	var validIDs []int32
-	for _, cand := range killCandidatees {
-		visibleIDs.Insert(cand.ID)
-		switch {
-		case cand.IsTerminal:
 			results = append(results, &apiv1.SearchActionResult{
-				Error: "",
-				Id:    cand.ID,
+				Error: fmt.Sprintf("Failed to kill search: %s", err),
```
Comment (AmanuelAaron) [inline master/internal/api_searches.go:228]: "Failed to cancel search"

File: master/internal/api_searches_intg_test.go [hunk 2]
```diff
@@ -367,6 +368,357 @@ func TestDeleteSearchesFilter(t *testing.T) {
...
+	require.Len(t, projHparam, 2)
+	require.Contains(t, projHparam, "test3.test4")
+	require.Contains(t, projHparam, "global_batch_size")
+
+	searchResp, err := api.SearchRuns(ctx, searchReq)
+	require.NoError(t, err)
+	require.Len(t, searchResp.GetRuns(), 2)
+}
+
+func TestKillSearchesNonTerminal(t *testing.T) {
```
Comment (AmanuelAaron) [inline master/internal/api_searches_intg_test.go:548]: In addition to this test case, we should add a test case for when some kills/cancels are successful and some fail.

Comment (jesse-amano-hpe) [reply master/internal/api_searches_intg_test.go:568]: Done :)

**Review by AmanuelAaron — APPROVED** [review]

**Pull request merged**